route/food-router.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const Food = require('../model/food.js');
+
+const foodRouter = module.exports = new Router();
+
+foodRouter.post('/api/food', jsonParser, function(req, res, next) {
+  req.body.timestamp = new Date();
+  new Food(req.body).save()
+  .then( food => res.json(food))
+  .catch(next);
+});
+
+foodRouter.get('/api/food/:id', function(req, res, next) {
+  Food.findById(req.params.id)
+  .then( food => res.json(food))
+  .catch(next);
+});
+
+foodRouter.put('/api/food/:id', jsonParser, function(req, res, next) {
+  Food.findByIdAndUpdate(req.params.id, { $set: { name: 'json bronn' }})
```

Comment:
I am not sure why you are using the `{ $set: { name: 'json bron' }}` here.

When making a put request what we want to do is update the the given object based on the ID with the req.body. Currently you are not accounting for the req.body in your update.